### PR TITLE
sf::Transform tutorial -- clarify transform order

### PR DESCRIPTION
**Pull request by Bromeon**
Clarify order of transforms, as highlighted by pvigier in https://github.com/SFML/SFML/issues/1608.

Once people agree with the wording, we can backport it to older tutorials.

### CODE REVIEW

File: tutorials/2.5/graphics-transform.php [hunk 1]
```diff
@@ -160,7 +160,7 @@
 sf::Transform t4 = t1 * t2 * t3;
 </code></pre>
 <p>
-    You can apply several predefined transformations to the same transform as well. They will all be combined sequentially:
+    You can apply several predefined transformations to the same transform as well. They will all be combined sequentially, in reverse order: the last operation (here <code>scale</code>) is applied first, and will be affected by operations above it in code (second would be <code>translate(-10.f, 50.f)</code>, for example).
```
Comment (eXpl0it3r) [inline tutorials/2.5/graphics-transform.php]: Is it always in reverse order or just reverse order for the listed provided here?

Personally, I still don't understand why it is applied in reverse. 🤔

Comment (Bromeon) [reply tutorials/2.5/graphics-transform.php]: Simple example: you start with a rectangle (black) as shown in the left graphic. The vector `v` represents each vertex' position -- the transformed rectangle is a result of transforming each vertex.
* First, you rotate `v` using the rotation matrix `R`. The result (blue) is `R * v`, which we call `w`.
* Now, you apply a translation `T` to `w`.  
Same as before, the result (green) is `T * w`, which is `T * (R * v)`.  
Matrix multiplication is associative; this is the same as `T * R * v`.
* Applying an additional scale transform `S` will yield `S * T * R * v` (red).

grafik

You see that even though the first transform was `R`, it's the rightmost expression in the equation -- or the last statement in code. In other words, `R` is written closest to `v`, and `R * v` can be understood as an intermediate vector, which is then further transformed by `T`, and last `S`.

Comment (Bromeon) [reply tutorials/2.5/graphics-transform.php]: I wonder if this topic (order of transform) is something that we should elaborate in the tutorial, or if it's out of scope.

Any other feedback to the PR?

**Comment by Bromeon** [comment]
Is this blocked by the missing translation?

**Comment by Bromeon** [comment]
Anyone wanting to translate this to French? :) 
(not sure it ends well if I try...)

Also, would it make sense to add something like this example to the doc?
https://github.com/SFML/SFML-Website/pull/135/files#r356254486

> Simple example: you start with a rectangle (black) as shown in the left graphic. The vector `v` represents each vertex' position -- the transformed rectangle is a result of transforming each vertex.
> * First, you rotate `v` using the rotation matrix `R`. The result (blue) is `R * v`, which we call `w`.
> * Now, you apply a translation `T` to `w`.  
> Same as before, the result (green) is `T * w`, which is `T * (R * v)`.  
> Matrix multiplication is associative; this is the same as `T * R * v`.
> * Applying an additional scale transform `S` will yield `S * T * R * v` (red).
> 
> grafik

**Comment by Bromeon** [comment]
Even if no one has the time/motivation to translate it to French, might be still worth adding it to the English docs.

**Comment by kimci86** [comment]
Hello,
I think it is wrong to say that operations (i.e. transformations) are combined (i.e. matrices are multiplied) in reverse order.
Transforming a vector with the combination of transformations T1 and T2 indeed is equivalent to first transforming the vector with T2 and then T1, but the combination still is in the given order T1 * T2.
Matrix multiplication is equivalent to function composition. The composition of functions g and f, noted g ∘ f, first applies f and then g when applied to an argument.

So I propose this wording (not sure when to use the word "transformation" or "operation"):
> You can apply several predefined transformations to the same transform as well. They will all be combined sequentially.
> Note that transforming an object with a combination a transformations is equivalent to applying each operation in reverse order. The last operation (here <code>scale</code>) is applied first, and will be affected by operations above it in code (second would be <code>translate(-10.f, 50.f)</code>, for example).

**Comment by Bromeon** [comment]
Adjusted it.

**Comment by kimci86** [comment]
In French, it can be:
> Remarquez que transformer un objet en combinant plusieurs transformations est équivalent à effectuer chaque opération dans l'ordre inverse. La dernière opération (ici <code>scale</code>) est appliquée en premier, et sera affectée par les opérations en amont dans le code (la deuxième serait <code>translate(-10.f, 50.f)</code>, par exemple).

**Comment by Bromeon** [comment]
Integrated @kimci86's translation, credited him via `Co-authored-by`.

**Review by eXpl0it3r — APPROVED** [review]

**Comment by Bromeon** [comment]
Can we merge this? 🙂

**Comment by eXpl0it3r** [comment]
Woops, forgot about this 🙂

**Pull request merged**